lib/connectors/memory.js

```diff
@@ -351,26 +351,54 @@ Memory.prototype.fromDb = function(model, data) {
   if (!data) return null;
   data = deserialize(data);
   var props = this._models[model].properties;
+  function cast(prop, val, props) {
```

Comment:
> The logic is still unique to the fromDb method and won't be used elsewhere.

Sure, that's fine. `fromDb` is called from a single method too, and yet it's a regular method.